backend/cmm_helpers.ml

```diff
@@ -1448,7 +1519,7 @@ let set_field_unboxed ~dbg memory_chunk block ~index_in_words newval =
     let field_address =
       array_indexing log2_size_addr block index_in_words dbg
     in
-    let newval = if size_in_bytes = 4 then low_32 dbg newval else newval in
+    let newval = low_bits newval ~dbg ~bits:(8 * size_in_bytes) in
```

Comment:
Not specific to this PR, but I noticed this does not write sign-extended int32#s into mixed blocks. Flambda2 tries to make sure the full content of blocks is deterministic (e.g. [in closures](https://github.com/ocaml-flambda/flambda-backend/blob/main/middle_end/flambda2/to_cmm/to_cmm_shared.ml#L398)), so I think [this code](https://github.com/ocaml-flambda/flambda-backend/blob/main/middle_end/flambda2/to_cmm/to_cmm_primitive.ml#L188) should probably be changed. But maybe this can never be used on static blocks so we don't care---we should check with the flambda2 devs.